test/srt/sampling/penaltylib/test_srt_endpoint_with_penalizers.py

```diff
@@ -0,0 +1,75 @@
+import json
+import unittest
+
+import requests
+
+from sglang.srt.utils import kill_child_process
+from sglang.test.test_utils import DEFAULT_MODEL_NAME_FOR_TEST, popen_launch_server
+
+
+class TestBatchPenalizerE2E(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.model = DEFAULT_MODEL_NAME_FOR_TEST
+        cls.base_url = f"http://localhost:{8157}"
```

Comment:
I don't recommend using localhost because it can cause issues in certain environments.
```
ERROR:    [Errno 99] error while attempting to bind on address ('::1', 8157, 0, 0): cannot assign requested address
```

Write it simply.
```suggestion
        cls.base_url = "http://127.0.0.1:8157"
```